tests/ui/macros/expr_2021_cargo_fix_edition.stderr

```diff
@@ -0,0 +1,33 @@
+error: the `expr` fragment specifier will accept more expressions in the 2024 edition.
+  --> $DIR/expr_2021_cargo_fix_edition.rs:6:9
+   |
+LL |     ($e:expr) => {
+   |         ^^^^
+   |
+   = warning: this changes meaning in Rust 2024
+   = note: for more information, see Migration Guide <https://doc.rust-lang.org/nightly/edition-guide/rust-2024/macro-fragment-specifiers.html>
+note: the lint level is defined here
+  --> $DIR/expr_2021_cargo_fix_edition.rs:3:9
+   |
+LL | #![deny(edition_2024_expr_fragment_specifier)]
+   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+help: to keep the existing behavior, use the `expr_2021` fragment specifier.
+   |
+LL |     ($e:expr_2021) => {
+   |         ~~~~~~~~~
+
+error: the `expr` fragment specifier will accept more expressions in the 2024 edition.
+  --> $DIR/expr_2021_cargo_fix_edition.rs:10:11
+   |
+LL |     ($($i:expr)*) => { };
+   |           ^^^^
+   |
+   = warning: this changes meaning in Rust 2024
+   = note: for more information, see Migration Guide <https://doc.rust-lang.org/nightly/edition-guide/rust-2024/macro-fragment-specifiers.html>
+help: to keep the existing behavior, use the `expr_2021` fragment specifier.
+   |
+LL |     ($($i:expr_2021)*) => { };
+   |           ~~~~~~~~~
```

Comment:
(Line picked semi-randomly...)

For this change, we should probably have one or more `run-rustfix` tests.  See:

https://rustc-dev-guide.rust-lang.org/tests/ui.html?highlight=rustfix#rustfix-tests